### PR TITLE
Minor refactor to how rate limits are handled and when

### PR DESCRIPTION
**Pull request by froi**
# Pull Request Template

Thank you for your contribution!

Before submitting this Pull Request, please make sure that a similar [Pull Request](https://github.com/gsa/code-gov-integrations/pulls) does not exist.

## Description

Minor refactor to how rate limits are handled and when. Parts of the projects were handling rate limits that made no sense. Sometimes handling the rate limit more than once for the same call. This added unneeded requests and execution time.

## Type of change

Please delete options that are not relevant.

- [x] Code cleanup (non-breaking change)

## How Has This Been Tested

Used the project's test suite:
image

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/github/utils.js [hunk 1]
```diff
@@ -76,26 +76,35 @@ async function handleRateLimit({ rateLimit, client }) {
  * @returns {Promise<{data: *, rateLimit: object}>}
  *
  * @example
- * let requestParams = { owner: 'gsa', repo: 'code-gov-integrations, state: 'open', labels: 'help wanted,code.gov', per_page: 10, page: 1};
+ * let requestParams = {
+ *   owner: 'gsa',
+ *   repo: 'code-gov-integrations,
```
Comment (saracope) [inline libs/github/utils.js]: Should have a closing quote here on line 81? @froi

Comment (froi) [reply libs/github/utils.js]: And that's what happens when I don't run the linter 😞 

Greate catch.

Comment (froi) [reply libs/github/utils.js]: I didn't notice it was in the JSDoc. I was really confused as to how the tests and other things passed 🤣

**Comment by froi** [comment]
@saracope Friendly ping to get this to your attention. 😄

**Comment by saracope** [comment]
Pinging @bjbhatt for review.

**Comment by froi** [comment]
Hello fine peeps. A friendly ping about this. I'd like to get it out of my PRs.

**Comment by bjbhatt** [comment]
Sure will  take a look at it tomorrow

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

File: libs/github/utils.js (hunk 1 above)
Comment (bjbhattGSA) [reply libs/github/utils.js]: I will merge this PR and update the missing quote.

**Pull request merged**